specs/binding-http.spec/src/main/resources/META-INF/zilla/http.idl

```diff
@@ -49,4 +49,77 @@ scope http
             HttpHeader[] headers;
         }
     }
+
+    scope event
+    {
+        enum HttpEventType (uint8)
+        {
+            CORS_PREFLIGHT_REQUEST_NOT_ALLOWED (1),
+            CORS_REQUEST_NOT_ALLOWED (2),
+            GUARD_AUTHORIZATION_FAILURE (3),
+            GUARD_AUTHORIZATION_SUCCESS (4),
+            ROUTE_NOT_FOUND (5),
+            ROUTE_FOUND (6)
+        }
+
+        enum Level
+        {
+            INFO,
+            WARNING,
+            ERROR
+        }
+
+        struct HttpEventCommon
+        {
+            int64 timestamp = 0;
+            int64 originId;
+            int64 routedId;
+            int64 initialId;
+            int64 replyId;
+            int64 affinity;
+            int64 authorization = 0;
+            int64 budgetId;
+            int64 traceId;
+        }
+
+        struct HttpEventCorsPreflightRequestNotAllowed extends HttpEventCommon
+        {
+            Level level = WARNING;
+        }
+
+        struct HttpEventCorsRequestNotAllowed extends HttpEventCommon
+        {
+            Level level = WARNING;
+        }
+
+        struct HttpEventGuardAuthorizationFailure extends HttpEventCommon
+        {
+            Level level = WARNING;
+        }
+
+        struct HttpEventGuardAuthorizationSuccess extends HttpEventCommon
+        {
+            Level level = INFO;
+        }
+
+        struct HttpEventRouteNotFound extends HttpEventCommon
+        {
+            Level level = WARNING;
+        }
+
+        struct HttpEventRouteFound extends HttpEventCommon
+        {
+            Level level = INFO;
+        }
+
+        union HttpEvent switch (HttpEventType)
+        {
+            case CORS_PREFLIGHT_REQUEST_NOT_ALLOWED: HttpEventCorsPreflightRequestNotAllowed corsPreflightRequestNotAllowed;
+            case CORS_REQUEST_NOT_ALLOWED: HttpEventCorsRequestNotAllowed corsRequestNotAllowed;
+            case GUARD_AUTHORIZATION_FAILURE: HttpEventGuardAuthorizationFailure guardAuthorizationFailure;
+            case GUARD_AUTHORIZATION_SUCCESS: HttpEventGuardAuthorizationSuccess guardAuthorizationSuccess;
+            case ROUTE_NOT_FOUND: HttpEventRouteNotFound routeNotFound;
+            case ROUTE_FOUND: HttpEventRouteFound routeFound;
+        }
```

Comment:
Then `HttpEvent` becomes an engine `Event` with `HttpEventType` type field.

Note: we may want a subtype for `AUTHORIZATION_SUCCESS` and `AUTHORIZATION_FAILURE` events to indicate the authorization identity. In both success and failure cases, they would use the same `HttpAuthorizationEvent` structure.

Given the generic nature of authorization events, we might also decide to define that at engine for reuse across different bindings, etc.

Reply:
not exactly sure what we want to record here... should we parse the jwt token and capture some values from there? or should we record the jwt token itself as a string? 🤔 pls LMK